user-manager/src/main/java/com/qpay/usermanager/service/impl/CustomerServiceImpl.java

```diff
@@ -9,51 +13,78 @@
+
     private final MerchantRepository merchantRepository;
 
+    private final AuthenticationClient authenticationClient;
+
     public CustomerEntity getCustomerById(final long id) {
         return customerRepository.findById(id).orElseThrow(CustomerNotFoundException::new);
     }
 
-    public CustomerEntity addCustomer(final CustomerModification customerModification) {
-        existsMerchantByEmail(customerModification.email());
-
-        var customerEntity = customerMapper.map(customerModification);
+    public CustomerEntity addCustomer(final CustomerCreation customerCreation) {
+        existsMerchantByEmail(customerCreation.email());
+        var customerEntity = customerMapper.map(customerCreation);
         try {
             customerEntity = customerRepository.save(customerEntity);
         } catch (final DataIntegrityViolationException e) {
             log.error(e.getMessage());
             throw new CustomerAlreadyExistsException(e.getMessage());
         }
+        final var newUser = userMapper.mapCustomerCreation(customerCreation);
+        authenticationClient.sendNewUser(newUser);
         return customerEntity;
     }
 
     public CustomerEntity updateCustomer(final CustomerModification customerModification, final long id) {
         existsMerchantByEmail(customerModification.email());
-
         final var customerEntity = getCustomerById(id);
+        final var previousEmail = customerEntity.getEmail();
         customerEntity.setEmail(customerModification.email());
         customerEntity.setName(customerModification.name());
         customerEntity.setPassword(customerModification.password());
-        return customerRepository.save(customerEntity);
+        final var savedCustomerEntity = customerRepository.save(customerEntity);
+        if (!previousEmail.equals(customerEntity.getEmail())) {
+            sendAuthenticationUserModification(customerModification, previousEmail);
+        }
+        return savedCustomerEntity;
     }
 
     public void deleteCustomer(final long id) {
+        final var customerEntity = getCustomerById(id);
         customerRepository.deleteById(id);
+        authenticationClient.sendToDeleteUser(customerEntity.getEmail());
+    }
+
+    private void sendAuthenticationUserModification(final CustomerModification customerModification, final String previousEmail) {
+        final UserCredentialsModification userCredentialsModification;
+        if (StringUtils.isNotEmpty(customerModification.password())) {
+            userCredentialsModification = userMapper.mapCustomerModification(customerModification);
+        } else {
+            userCredentialsModification = UserCredentialsModification.builder()
+                    .email(customerModification.email())
+                    .build();
+        }
+        authenticationClient.sendUpdatedUser(userCredentialsModification, previousEmail);
     }
 
     private void existsMerchantByEmail(final String email) {
```

Comment:
Fiasco. You have to check customer repository, not merchant.
 I cannot check every single line. You have to review code as well, not mark approve all the time

Reply:
it's correct, because here we need to check, that merchant with this email doesn't exist when we try to add cutomer with this email